File: frontend/lib/models/app_view_model.dart
import 'permission.dart';

class AppViewModel {
  final List<Permission> permissions;
  final bool isPaid;
  final String url;
  final String imageUrl;
  final String type;
  final String id;
  final String name;
  final String developer;
  final double averageRating;
  final int reviewCount;
  final String releaseDate;

  AppViewModel({
    required this.permissions,
    required this.isPaid,
    required this.url,
    required this.imageUrl,
    required this.type,
    required this.id,
    required this.name,
    required this.developer,
    required this.averageRating,
    required this.reviewCount,
    required this.releaseDate
  });

  factory AppViewModel.fromJson(Map<String, dynamic> json) {
    return AppViewModel(
      permissions: json['permissions'] != null 
          ? (json['permissions'] as List).map((p) => Permission.fromJson(p)).toList()
          : [],
      isPaid: json['isPaid'] as bool,
      url: json['url'] as String,
      imageUrl: json['imageUrl'] as String,
      type: json['type'] as String,
      id: json['id'] as String,
      name: json['name'] as String,
      developer: json['developer'] as String,
      averageRating: (json['averageRating'] as num?)?.toDouble() ?? 0.0,
      reviewCount: json['reviewCount'] as int? ?? 0,
      releaseDate: json['releaseDate'] as String
    );
  }
}

File: backend/src/GarminFilter.Api/Models/AppViewModel.cs
using GarminFilter.Domain.App.Aggregates;
using GarminFilter.SharedKernel.App.ValueObjects;

namespace GarminFilter.Api.Models;

public class AppViewModel
{
	public AppViewModel(AppAggregate app)
	{
		Id = app.Id;
		Name = app.Name;
		Type = app.Type;
		ImageUrl = $"/app/{app.IconFileId}";
		Url = $"https://apps.garmin.com/apps/{app.Id}";
		IsPaid = app.IsPaid;
		Permissions = app.RequiredPermissions
			.Select(permission => new AppPermissionViewModel(permission))
			.OrderBy(permission => permission.Description)
			.ToList();
		Developer = app.DeveloperName;
		AverageRating = app.AverageRating;
		ReviewCount = app.ReviewCount;
		ReleaseDate = DateTime.UnixEpoch.AddMilliseconds(app.ReleaseDate).ToString("MMM dd, yyyy");
	}

	public string ReleaseDate { get; }

	public uint ReviewCount { get; }

	public decimal AverageRating { get; }

	public string Developer { get; }


	public List<AppPermissionViewModel> Permissions { get; }

	public bool IsPaid { get; }

	public string Url { get; }

	public string ImageUrl { get; }

	public AppType Type { get; }

	public AppId Id { get; }
	public string Name { get; }
}